src/components/Dragons/Dragons.js

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Dragon from './Dragon';
+import '../../styles/dragons.css';
+import { fetchDragons } from '../../redux/dragons';
+
+const Dragons = () => {
+  const dispatch = useDispatch();
+  const shouldFetch = useRef(true);
+  const dragonsArr = useSelector((state) => state.dragons);
+
+  useEffect(() => {
+    if (shouldFetch.current) {
+      if (dragonsArr.length === 0) {
+        shouldFetch.current = false;
+        dispatch(fetchDragons());
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
- [ ] Kindly avoid disabling linters, instead, I suggest you work around them since it's a project requirement in every Microverse project and they also enforce a clean code that uses best practices 👍 (this applies to all the other files too)